Enable the `systemcrypto` goexperiment by default (#1698)
microsoft/go: eng/doc/fips/README.md

>
> 1.24 introduces `GODEBUG=fips140=on` as the preferred way to enable FIPS mode. See also [the Go 1.24 changelog](#go-124-feb-2025).

> [!NOTE]

> > Since Go 1.25 `systemcrypto` is enabled by default on Windows and Linux, there is no need to set it anymore. See also [the Go 1.25 changelog](#go-125-aug-2025).

Reviewer: ```suggestion
> Since Go 1.25, `systemcrypto` is enabled by default on Linux and Windows. There is no need to manually enable using OpenSSL/CNG under the hood anymore. See also [the Go 1.25 changelog](#go-125-aug-2025).
```

Fix missing comma and comma splice, and it also might be interesting to borrow language from the Microsoft crypto policy for a reader who got here from there. (So, rearrange platforms to match.)

The `GOEXPERIMENT` environment variable is used at build time to select a cryptographic library backend. This modifies the Go runtime included in the program to use the specified platform-provided cryptographic library whenever it calls a Go standard library crypto API. The `GOEXPERIMENT` values that pick a crypto backend are:

- `systemcrypto` automatically selects the suggested crypto backend for the target platform
   - Prior to Go 1.21, this experiment is not available and the backend must be selected manually

>    - Since Go 1.25, this experiment es enabled automatically on Windows and Linux. It can be disabled in the usual way: setting `GOEXPERIMENT=nosystemcrypto`

Reviewer: ```suggestion
   - Since Go 1.25, this experiment is enabled automatically on Windows and Linux. It can be disabled like any other `GOEXPERIMENT`: setting `GOEXPERIMENT=nosystemcrypto`
```

Trying to answer the Q: "in the usual way *for what*?"

This could also be moved to "Usage: GOEXPERIMENTs and backend build tags", potentially.

### Dockerfile env instruction

> [!NOTE]

> > Since Go 1.25 `systemcrypto` is enabled by default on Windows and Linux, there is no need to set it anymore. See also [the Go 1.25 changelog](#go-125-aug-2025).

Reviewer: ^

### Modify the build command

> [!NOTE]

> > Since Go 1.25 `systemcrypto` is enabled by default on Windows and Linux, there is no need to set it anymore. See also [the Go 1.25 changelog](#go-125-aug-2025).

Reviewer: ^

### Go 1.25 (Aug 2025)

> - Enable the `systemcrypto` goexperiment by default behavior on Windows and Linux. To disable it, set `GOEXPERIMENT=nosystemcrypto`.

Reviewer: Wording seems out of place:

```suggestion
- The `systemcrypto` goexperiment is now enabled by default on Windows and Linux. To disable it, set `GOEXPERIMENT=nosystemcrypto`.
```
